cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -321,6 +331,14 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 
 	// IsEmptyStatus is to check if initial status of cVR object is empty.
 	if IsEmptyStatus(cVR) || IsInitStatus(cVR) || IsRecreateStatus(cVR) {
+		// We should generate replicaID for new volume only
+		if IsEmptyStatus(cVR) || IsInitStatus(cVR) {
+			if err := volumereplica.GenerateReplicaID(cVR); err != nil {
+				klog.Errorf("cVR ReplicaID creation failure: %v", err.Error())
+				return string(apis.CVRStatusOffline), err
+			}
+		}
+
```

Comment:
Can we have validation for replicaId in case of recreate phase. If it is empty we will never allow creating volumes

Reply:
ok